### ISSUE TITLE
Configurable Cache Control

### ISSUE THREAD

**Issue by wycats** [static-web-server/static-web-server#30]
I'm using `static-web-server` as a development server, and I would love the ability to configure the `Cache-Control` headers in some way.

Some sort of flag would be fine, but have you considered configuration beyond flags in general? Do you already have thoughts on the topic?

**Comment by joseluisq** [comment]
That's razonable, I have no suggestions right now but open to some ideas so feel free to suggest something or send a PR.

**Comment by joseluisq** [comment]
Just to share an idea while I came across with the existing cache control feature.
For me the easiest way to kind of workaround part of the current limitation is using two new flags like:

```sh
# To configure the extensions to apply the header 
--cache-control-extensions="atom,json,rss,xml"
# To configure the max-age
--cache-control-expires=2h
```

The problem is that you could cache certain files in one-way only.

I recognize that the existing feature is arbitrary and ad-hoc however the possibility of a config file is also not discarded but of course implies more extra work.

Thoughts?

**Comment by joseluisq** [comment]
Just to "unfreeze" this feature request.
In **v2** you can disable or enable the headers via `-e=true|false, --cache-control-headers=true|false`.
However yeah we need something more flexible so I will plan something around it like a proper config file (probably a .toml one) because it can also help for example defining custom routes which I am also interested in.

**Comment by joseluisq** [comment]
Config file setup is under proposal on https://github.com/joseluisq/static-web-server/discussions/69. Feedback welcome!

**Comment by Radiergummi** [comment]
I just hit this (CSP in my case), and was wondering whether it wouldn't be easier to allow string values for cache control headers and security headers, so you could set your own version if the defaults are no good for a specific use case? Maybe just a newline-delimited list of headers...

**Comment by joseluisq** [comment]
The new PR #101 address this feature request via a new config file as also explained in https://github.com/joseluisq/static-web-server/issues/100#issuecomment-1116131506.

**Comment by joseluisq** [comment]
Released on [v2.8.0](https://github.com/joseluisq/static-web-server/releases/tag/v2.8.0)
